torch/_inductor/lowering.py

```diff
@@ -5119,6 +5119,82 @@ def all_to_all_single(
             )
         )
 
+    _c10d_functional = torch.ops._c10d_functional
+
+    @register_lowering(_c10d_functional.all_reduce)
+    def _all_reduce(inp, reduce_op, group_name):
+        return ir.TensorBox.create(
+            ir._CollectiveKernel.create_inplace_single(
```

Comment:
I don't quite understand these semantics. The underlying aten op is out of place right? Why can we just make it inplace here?

Reply:
> The underlying aten op is out of place right?

The underlying op is in-place (`all_reduce_`).